python/lsst/afw/table/baseRecord.cc

```diff
@@ -22,16 +22,81 @@
 
 #include <pybind11/pybind11.h>
 //#include <pybind11/operators.h>
-//#include <pybind11/stl.h>
+#include <pybind11/stl.h>
+
+#include "numpy/arrayobject.h"
+#include "ndarray/pybind11.h"
+#include "ndarray/converter.h"
+
+#include "lsst/afw/table/Flag.h"
+#include "lsst/afw/table/Field.h"
+#include "lsst/afw/table/SchemaMapper.h"
+#include "lsst/afw/table/BaseColumnView.h"
+#include "lsst/afw/table/BaseRecord.h"
 
 namespace py = pybind11;
 
 using namespace lsst::afw::table;
 
+template <typename T>
+void declareBaseRecordOverloads(py::class_<BaseRecord, std::shared_ptr<BaseRecord>> clsBaseRecord,
+                                const std::string & suffix){
+    clsBaseRecord.def(("_get_"+suffix).c_str(),
+                      (typename Field<T>::Value (BaseRecord::*)(Key<T> const &) const) &BaseRecord::get);
+    clsBaseRecord.def("_getitem_", [](BaseRecord & baseRecord, Key<T> const & key)->typename Field<T>::Reference{
+        /*
+        Define the python __getitem__ method in python to return a baserecord for the requested key
```

Comment:
Just to be sure I understand: @TallJimbo is suggesting changing `Key<T> const & key)` to `Key<T> const & self)`? (C++ lambdas are still quite new to me).